Tools/autotest/common.py

```diff
@@ -13488,9 +13501,41 @@ def autotest(self, tests=None, allow_skips=True):
                 print(str(failure))
 
         self.post_tests_announcements()
+        if self.generate_junit:
+            if step_name is None:
+                step_name = "Unknown_step_name"
+            step_name.replace(".", "_")
+            self.create_junit_report(step_name, results, skip_list)
 
         return len(self.fail_list) == 0
 
+    def create_junit_report(self, test_name: str, results: List[Result], skip_list: list[tuple[Test, dict[str, str]]]) -> None:
+        """Generate Junit report from the autotest results"""
+        from junitparser import TestCase, TestSuite, JUnitXml, Skipped, Failure
+        frame = self.vehicleinfo_key()
+        xml_filename = f"autotest_result_{frame}_{test_name}_junit.xml"
+        self.progress(f"Writing test result in jUnit format to {xml_filename}\n")
+
+        suites = TestSuite("ArduPilot Autotest")
```

Comment:
Why the two test suites?

Reply:
first one create the testsuites

Reply:
This one is empty except for a timestamp?
```python
        suites = TestSuite("ArduPilot Autotest")
        suites.timestamp = datetime.now().replace(microsecond=0).isoformat()
```

This one gets the test cases added to it?
```python
        suite = TestSuite(f"Autotest {frame} {test_name}")
```

Reply:
yes
the first one is the equivalent of your  : f.write('<testsuites name="Autotest" timestamp="%s">\n\n' % now.isoformat())

Reply:
Except they're not equivalent:
* I'm putting the timestamp on the top-level `<testsuites>` element, e.g.
    ```xml
    <testsuites name="ArduPilot Autotest" timestamp="2023-10-23T10:12:07">
        <testsuite name="Autotest Frame Test">...</testsuite>
    </testsuites>
    ```
* You're putting the timestamp on a second `<testsuite>` element, e.g.
    ```xml
    <testsuites>
        <testsuite name="ArduPilot Autotest" timestamp="2023-10-23T10:12:07"/>
        <testsuite name="Autotest Frame Test">...</testsuite>
    </testsuites>
    ```

Reply:
ha yes... doesn't work as intent ... from the lib code the first call create the testsuites, but it doesn't do it correctly ... annoying